### DISCUSSION TITLE
Strange popup in Next app

### DISCUSSION THREAD

**Discussion by kylemclaren** [vercel/next.js#45873] [Help] [answered]
I am developing a Next app and I have noticed a strange popup in the bottom right corner of the page when transitioning between routes. It flashes by so quickly that I am having trouble debugging it to find what might be loading it. I have looked in the dev console and even recorded a HAR archive but neither of those seem to show anything out of the ordinary.

I can reproduce this quite reliably by starting a fresh dev server (`next dev`), opening up localhost, then transitioning between routes. After a few clicks it does not pop up again and I need to restart the dev server to see the popup again. I have only tested this locally and have not pushed the app to Vercel etc to see if the behaviour is present there.

https://user-images.githubusercontent.com/3727384/218579938-1e955c90-9b2e-46c8-8f3e-c1e579af3659.mp4

Hoping someone in the community recognizes what is happening here. Apologies for my rudimentary explanation of this issue.

**Comment by ciruz** [comment]
This is just the next.js animated indicator, for fast refresh / hot reloading.

**Comment by kfirfitousi** [comment, marked as answer]
It's the Next.js Fast Refresh in action.  
When running `next dev`, Next.js isn't building any of the routes. Instead, it build them on demand when you navigate to them.  
You can see this in the server console:

```
wait  - compiling /page (client and server)...
```
The popup may be shown during this build and on further builds (i.e. when editing the file).

**Reply by kylemclaren** [reply]
Excellent, that makes perfect sense. Thank you.
